fdbcli/fdbcli.actor.h

```diff
@@ -47,8 +47,28 @@ struct CommandHelp {
 	CommandHelp(const char* u, const char* s, const char* l) : usage(u), short_desc(s), long_desc(l) {}
 };
 
+void arrayGenerator(const char* text, const char* line, const char** options, std::vector<std::string>& lc);
+
 struct CommandFactory {
-	CommandFactory(const char* name, CommandHelp help) { commands()[name] = help; }
+	typedef void (*CompletionGeneratorFunc)(const char* text,
+	                                        const char* line,
+	                                        std::vector<std::string>& lc,
+	                                        std::vector<StringRef> const& tokens);
```

Comment:
Do we need the `std::vector<StringRef> const& tokens` here, which is not used

Reply:
The generator for the `throttle` command makes use of them. Basically we need this if the sequence of options can change depending on what arguments we've entered.

Reply:
Oh, cool. I see it, thanks for the clarification.